self_testim.py

```diff
@@ -0,0 +1,41 @@
+def is_valid(board, neighbor, initial_point):  # neighbor[0]=row index, neighbor[1]=col index,
+    return 0 <= neighbor[0] < len(board) and 0 <= neighbor[1] < len(board[neighbor[0]]) \
```

Comment:
Add function `is_valid_point`, don't write the same code twice (0 <= x < len(board))

Reply:
Change `len(board[neighbor[0]])` to `len(board[0])`